atlasdb-perf/src/main/java/com/palantir/atlasdb/performance/benchmarks/KvsGetCandidateCellsForSweepingBenchmarks.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.performance.benchmarks;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweeping;
+import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweepingRequest;
+import com.palantir.atlasdb.keyvalue.api.ImmutableCandidateCellForSweepingRequest;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.performance.benchmarks.table.ConsecutiveNarrowTable;
+import com.palantir.common.base.ClosableIterator;
+
+@State(Scope.Benchmark)
+public class KvsGetCandidateCellsForSweepingBenchmarks {
+
+    @Benchmark
+    @Threads(1)
+    @Warmup(time = 20, timeUnit = TimeUnit.SECONDS)
```

Comment:
I ran the benchmarks using our orchestration and here are the results:

## Postgres
> `fullTableScanCleanConservative` 
> `16 samples, 10427039.744 ± 112795.694 us/op`
> `fullTableScanCleanThorough`
> `9 samples,  18466122.411 ±  97663.498 us/op`
> `fullTableScanDirtyConservative`
> `36 samples, 4473458.233 ±  14631.149 us/op`
> `fullTableScanDirtyThorough`
> `31 samples, 5239362.197 ±  17299.716  us/op`

## Cassandra
> `fullTableScanCleanConservative` 
> `15 samples, 11021512.431 ±   144994.974  us/op`
> `fullTableScanCleanThorough`
> `8 samples, 20673724.416 ±   186058.102  us/op`
> `fullTableScanDirtyConservative`
> `32 samples, 5052694.528 ±  1180736.681  us/op`
> `fullTableScanDirtyThorough`
> `8 samples, 21875392.512 ± 10290446.479  us/op`

## Oracle
> `fullTableScanCleanConservative` 
> `5 samples, 33064537.293 ± 2286985.705  us/op`
> `fullTableScanCleanThorough`
> `5 samples, 36628017.971 ± 1784729.646  us/op`
> `fullTableScanDirtyConservative`
> `16 samples, 10241441.792 ±  203461.894  us/op`
> `fullTableScanDirtyThorough`
> `18 samples, 9232129.138 ±  159976.107  us/op`

In general, the results are consistent enough except for fullTableScanDirtyThorough on Cassandra. I would still like to run these on smaller tables, to get in more repetitions and try to avoid adding 10+ minutes to the overall benchmark runtime, but I am fine merging this in and reevaluating later.